pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -90,6 +90,27 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1alp
 		return gardencorev1alpha1helper.LastError(fmt.Sprintf("Failed to retrieve the Shoot namespace in the Seed cluster (%s)", err.Error()))
 	}
 
+	// Unregister the Shoot as Seed cluster if it was annotated to be a seed and is in the garden namespace
+	if o.Shoot.Info.Namespace == common.GardenNamespace && o.ShootedSeed != nil {
+		if err := botanist.UnregisterAsSeed(); err != nil {
+			return gardencorev1alpha1helper.LastError(fmt.Sprintf("Could not unregister Shoot %q as Seed: %+v", o.Shoot.Info.Name, err))
+		}
+
+		// wait seed to be deleted before to go on with the shoot
+		if err = utilretry.UntilTimeout(context.TODO(), time.Second, 10*time.Second, func(context.Context) (done bool, err error) {
```

Comment:
```suggestion
		if err := utilretry.UntilTimeout(context.TODO(), time.Second, 10*time.Second, func(context.Context) (done bool, err error) {
```

Is `10s` enough as wait time for a regular seed deletion?

Reply:
I forgot to increase it before to push, now it is set to `300s`.